packages/neon_framework/lib/src/storage/sqlite_cookie_persistence.dart

```diff
@@ -0,0 +1,296 @@
+  ///
+  /// Optionally [allowedBaseUri] can be used to restrict storage and loading of cookies to a certain domain and path.
+  SQLiteCookiePersistence({
+    required this.accountID,
+    this.allowedBaseUri,
+  });
+
+  final _log = Logger('SQLiteCookiePersistence');
+
+  /// The allowed cookie base uri.
+  ///
+  /// When not null, only cookies  domain and path matching this uri will be
+  /// persisted.
+  ///
+  /// See:
+  ///   * domain matching [RFC6265 section 5.1.3](https://httpwg.org/specs/rfc6265.html#rfc.section.5.1.3)
+  ///   * path matching [RFC6265 section 5.1.4](https://httpwg.org/specs/rfc6265.html#rfc.section.5.1.4)
+  final Uri? allowedBaseUri;
+
+  /// The id of the requesting account.
+  final String accountID;
+
+  static Database? _database;
+
+  /// Initializes this cookie persistence by setting up the backing SQLite database.
+  ///
+  /// This must called and completed before accessing other methods of the cache.
+  static Future<void> init() async {
+    if (_database != null) {
+      return;
+    }
+
+    assert(
+      NeonPlatform.instance.canUsePaths,
+      'Tried to initialize SQLiteCookiePersistence on a platform without support for paths',
+    );
+    final cacheDir = await getApplicationCacheDirectory();
+    _database = await openDatabase(
+      p.join(cacheDir.path, 'cookies.db'),
+      version: 1,
+      onCreate: onCreate,
+    );
+  }
+
+  @visibleForTesting
+  static Future<void> onCreate(Database db, int version) async {
+    // https://httpwg.org/specs/rfc6265.html#storage-model with an extra account key.
+    await db.execute('''
+PRAGMA case_sensitive_like = true;
+
+CREATE TABLE "cookies" (
+  "account"           TEXT NOT NULL,
+	"name"              TEXT NOT NULL,
+	"value"             TEXT NOT NULL,
+	"expiry-time"       INTEGER NOT NULL,
+	"domain"            TEXT NOT NULL,
+	"path"              TEXT NOT NULL,
+	"creation-time"     INTEGER NOT NULL,
+	"last-access-time"  INTEGER NOT NULL,
+	"persistent-flag"   BOOLEAN NOT NULL,
+	"host-only-flag"    BOOLEAN NOT NULL,
+	"secure-only-flag"  BOOLEAN NOT NULL,
+	"http-only-flag"    BOOLEAN NOT NULL,
+  PRIMARY KEY("account","name","domain","path")
+);
+
+CREATE TRIGGER delete_outdated_cookies
+   BEFORE UPDATE 
+   ON cookies
+BEGIN
+ DELETE FROM cookies
+ WHERE
+   datetime("expiry-time", 'unixepoch', 'utc') <= datetime('now', 'utc');
+END;
+''');
+  }
+
+  Database get _requireDatabase {
+    final database = _database;
+    if (database == null) {
+      throw StateError(
+        'Cookie persistence has not been set up yet. Please make sure SQLiteCookiePersistence.init() has been called before and completed.',
+      );
+    }
+
+    return database;
+  }
+
+  @override
+  Future<List<Cookie>> loadForRequest(Uri uri) async {
+    final list = <Cookie>[];
+
+    if (allowedBaseUri != null) {
+      if (!isDomainMatch(allowedBaseUri!.host, uri.host) || !isPathMatch(allowedBaseUri!.path, uri.path)) {
+        return list;
+      }
+    }
```

Comment:
```suggestion
    if (allowedBaseUri != null) {
      if (!isDomainMatch(allowedBaseUri!.host, uri.host) || !isPathMatch(allowedBaseUri!.path, uri.path)) {
        return [];
      }
    }
    
    final list = <Cookie>[];
```
This will avoid that any additional code added later could already modify the list and then some cookies would be returned.